direct/core/src/main/java/cz/o2/proxima/direct/MyLogObserver.java

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright 2017-2021 O2 Czech Republic, a.s.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package cz.o2.proxima.direct;
+
+import cz.o2.proxima.annotations.Stable;
+import cz.o2.proxima.storage.Partition;
+import cz.o2.proxima.storage.StreamElement;
+import cz.o2.proxima.time.WatermarkSupplier;
+import java.io.Serializable;
+
+/** Base interface for bulk and online observers. */
+@Stable
+public interface MyLogObserver<T extends MyLogObserver.OnNextContext> extends Serializable {
```

Comment:
Probably should not be Serializable. If we have still some parts, that are Serializable, it would be nice to take this opposrtunity to clear that up as well.

Reply:
Agreed, but that's a different issue and is out of scope of this PR. (I don't have enough time to do both and it's not a problem to solve this one later).